glean-core/csharp/Glean/LibGleanFFI.cs

```diff
@@ -240,6 +240,45 @@ bool disabled
         [DllImport(SharedGleanLibrary, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
         internal static extern byte glean_uuid_test_has_value(UuidMetricTypeHandle metric_id, string storage_name);
 
+        // Custom pings
+
+        /// <summary>
+        /// A handle for the ping metric type, which performs cleanup.
+        /// </summary>
+        internal sealed class PingTypeHandle : BaseGleanHandle
+        {
+            protected override bool ReleaseHandle()
+            {
+                if (!this.IsInvalid)
+                {
+                    glean_destroy_ping_type(handle);
+                }
+
+                return true;
+            }
+        }
+
+        [DllImport(SharedGleanLibrary, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern PingTypeHandle glean_new_ping_type(
+            string name,
+            byte include_client_id,
+            byte send_if_empty,
+            string[] reason,
+            Int32 reason_codes_len
+        );
+
+        [DllImport(SharedGleanLibrary, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern void glean_destroy_ping_type(IntPtr handle);
+
+        [DllImport(SharedGleanLibrary, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern StringAsReturnValue glean_ping_collect(PingTypeHandle ping_type_handle, string reason);
```

Comment:
Sorry to be nitpicky on this, but would you kindly move this function to the same position it has in the Kotlin implementation [here](https://searchfox.org/glean/rev/cfabeff6ba9c3562332dbe8640f16f26e99f0a46/glean-core/android/src/main/java/mozilla/telemetry/glean/rust/LibGleanFFI.kt#112)?

This belongs to the Glean General API, not to the Ping Metric type specific API.

Also, please note that `glean_submit_ping_by_name` is missing :)